api/java/src/main/java/com/qbot/answeringservice/service/UserService.java

```diff
@@ -42,10 +73,26 @@ public User updateUser(User user) {
         return userRepo.save(user);
     }
 
-    public void deleteUser(UUID userId) {
+    public void deleteUser(String userId) {
         if (userId != null) {
             userRepo.deleteById(userId);
         }
     }
 
+    private String validateUserRequest(UserRequest userRequest) {
+        StringBuilder builder = new StringBuilder();
+
+        if (userRequest.getEmailAddress() == null
+                || !EmailValidator.getInstance().isValid(userRequest.getEmailAddress())) {
```

Comment:
There's a chance for an NPE here - can we make it so that we have EmailValidator autowired in or just have it be a non-singleton and just have a validation method + have a null check prior to this?

Reply:
done